core/src/test/java/org/elasticsearch/routing/AliasRoutingIT.java

```diff
@@ -40,13 +48,111 @@
 /**
  *
  */
-public class AliasRoutingTests extends ESIntegTestCase {
+public class AliasRoutingIT extends ESIntegTestCase {
+
+    public static class MockScriptPlugin extends Plugin {
+
+        public MockScriptPlugin() {
+        }
+
+        @Override
+        public String name() {
+            return MockScriptEngine.NAME;
+        }
+
+        @Override
+        public String description() {
+            return "Mock script engine for " + AliasRoutingIT.class;
+        }
+
+        public void onModule(ScriptModule module) {
+            module.addScriptEngine(MockScriptEngine.class);
+        }
+
+    }
+
+    public static class MockScriptEngine implements ScriptEngineService {
```

Comment:
I don't think it needs to be abstract, but I think you should inherit from `org.elasticsearch.script.MockScriptEngine`. Or refactor that to allow more custom logic (right now search/execute do nothing).